src/pocketmine/command/defaults/Threads/TimingsPasteTask.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+namespace pocketmine\command\defaults\Threads;
+
+use pocketmine\scheduler\AsyncTask;
+use pocketmine\Server;
+
+class TimingsPasteTask extends AsyncTask{
+
+	protected $name;
+	protected $data;
+	protected $serverName;
+	protected $version;
+
+	protected $messageParams;
+	protected $failed = false;
+
+	public function __construct($name, $data, $serverName, $version){
+		$this->name = $name;
+		$this->data = unserialize($data);
```

Comment:
why serializing/unserializing here? you're still on the main thread in the constructor. You should be _serializing_ here, not unserializing. An AsyncTask is not a thread.